File: BaseLibrary/src/main/java/com/sprout/baselibrary/base/BasePresenter.java
package com.sprout.baselibrary.base;

 public abstract class BasePresenter<V extends BaseView, M extends BaseModel> {
    protected V mView;
    protected M mModel;

    void attachView(V view) {
        this.mView = view;
        this.mModel = getModel();
    }

    protected abstract M getModel();

    void detachView() {
        mView = null;
        mModel = null;
    }
}

File: BaseLibrary/src/main/java/com/sprout/baselibrary/base/BasePresenter.kt
package com.sprout.baselibrary.base

import android.view.View

abstract class BasePresenter<V : BaseView, M : BaseModel> {
    var mView: V? = null
    var mModel: M? = null
    fun attachView(view: V) {
        mView = view
        mModel = getModel()
    }

    abstract fun getModel(): M

    fun detachView() {
        mView = null
        mModel = null
    }
}